include/tvm/relay/op_attr_types.h

```diff
@@ -81,10 +81,16 @@ using TOpIsStateful = bool;
  */
 using TNonComputational = bool;
 
+enum ShapeDependantKind {
+  kShapeDependantShape = 0,
+  kShapeDependantData = 1,
+  kShapeDependantBoth = 2,
```

Comment:
No need to have `kShapeDependantBoth`. When you provide the data tensor as input to the shape function, you can retrieve its shape directly. So it's unnecessary to provide both shape and data tensor.

Reply:
Thanks for you comment.

I tried two ways to use data tensor's shape directly.
First, I passed data tensor's shape to _reshape_shape_func but I got this error
```
ValueError: All indices are supposed to be constants
```
Second, I tried to construct a tensor by tvm.te.placeholder based on data tensor's shape, but it doesn't work because it can't be passed to tvm.lower called later.

Reply:
@icemelon9 This is solved, could you please take another look?